clippy_lints/src/matches/match_same_arms.rs

```diff
@@ -103,17 +104,21 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, arms: &'tcx [Arm<'_>]) {
     let indexed_arms: Vec<(usize, &Arm<'_>)> = arms.iter().enumerate().collect();
     for (&(i, arm1), &(j, arm2)) in search_same(&indexed_arms, hash, eq) {
         if matches!(arm2.pat.kind, PatKind::Wild) {
-            span_lint_and_then(
-                cx,
-                MATCH_SAME_ARMS,
-                arm1.span,
-                "this match arm has an identical body to the `_` wildcard arm",
-                |diag| {
-                    diag.span_suggestion(arm1.span, "try removing the arm", "", Applicability::MaybeIncorrect)
-                        .help("or try changing either arm body")
-                        .span_note(arm2.span, "`_` wildcard arm here");
-                },
-            );
+            if !cx.tcx.features().non_exhaustive_omitted_patterns_lint
```

Comment:
It should probably be a good idea to add a `FIXME` comment here for when `non_exhaustive_omitted_patterns` gets stabilized [(rust#89554)](https://github.com/rust-lang/rust/issues/89554). That would make Clippy crash AFAIK

Note that this should only be removed if Clippy is being compiled for Rust with that feature. So, unless we have  some magic to check which version is Clippy being installed (or some other methods)

Or we can just not have this stress and just use `cx.tcx.features().active(sym::non_exhaustive_omitted_patterns_lint)`

I'm not sure what branch is the best to follow here.

```suggestion
            // FIXME(Centri3): ...
            if !cx.tcx.features().non_exhaustive_omitted_patterns_lint
```

Reply:
I'll do the latter in a bit, but I'm curious then, why is this used so much? (`cx.tcx.features().<field>`) I haven't seen `.active` before.

Reply:
There's no ICE risk here. What will happen is the stabilization pr will remove the field which will cause clippy to fail to compile. Since clippy is in the rust repo the same pr will also fix clippy by removing condition.

Basically this doesn't need a note in the source code.